src/main/scala/eu/proteus/solma/lasso/algorithm/FlatnessMappingAlgorithm.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2017 The Proteus Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package eu.proteus.solma.lasso.algorithm
+
+/**
+  * TODO description
+  *
+  */
+class FlatnessMappingAlgorithm(input: List[Double],
```

Comment:
I would suggest to check if the data structures used here are cheap enough w.r.t to the workload (e.g., index based lookups cost O(n) for a Scala List).